crates/rethnet_evm/src/db/remote.rs

```diff
@@ -0,0 +1,75 @@
+use tokio::runtime::{Builder, Runtime};
+
+use rethnet_eth::remote::RpcClient;
+use rethnet_eth::{Address, B256, U256};
+
+use revm::{db::DatabaseRef, AccountInfo, Bytecode};
+
+#[derive(std::hash::Hash, std::cmp::Eq, std::cmp::PartialEq)]
+struct StorageLocation {
+    address: Address,
+    index: U256,
+}
+
+struct RemoteDatabase {
+    client: RpcClient,
+    runtime: Runtime,
+}
+
+impl RemoteDatabase {
+    pub fn _new(url: &str) -> Self {
+        Self {
+            client: RpcClient::new(url),
+            runtime: Builder::new_multi_thread()
+                .build()
+                .expect("failed to construct async runtime"),
+        }
+    }
+}
+
+impl DatabaseRef for RemoteDatabase {
+    type Error = anyhow::Error;
```

Comment:
Please make this a dedicated error for `RemoteDatabase` using `thiserror`. I'm making the same changes for the other places I was originally using `anyhow::Error`

Reply:
I like it, thanks, done in the latest version of my commit.